lib/connectors/memory.js

```diff
@@ -255,6 +255,21 @@ Memory.prototype._createSync = function(model, data, fn) {
     this.collection(model, {});
   }
 
+  var indexes = this._models[model].model.definition.indexes();
+  for (var idx in indexes) {
+    var idxPropName = idx.substring(0, idx.indexOf('_index'));
+    if (indexes[idx].unique === true) {
+      for (var instId in this.cache[model]) {
+        var inst = JSON.parse(this.cache[model][instId]);
```

Comment:
The reason I have to use JSON parse is because we stringify the model instances to store in the cache :(. However, I like that approach and I'm trying it out. I think it will work well with multiple-key unique indexes too.